s4e-backend/src/main/java/pl/cyfronet/s4e/service/InstitutionService.java

```diff
@@ -29,65 +25,91 @@
 @Slf4j
 public class InstitutionService {
     private final InstitutionRepository institutionRepository;
-    private final AppUserRepository appUserRepository;
     private final UserRoleService userRoleService;
     private final SlugService slugService;
     private final FileStorageProperties fileStorageProperties;
     private final FileStorage fileStorage;
+    private final InstitutionMapper institutionMapper;
 
     @Transactional(rollbackFor = InstitutionCreationException.class)
-    public String save(CreateInstitutionRequest request) throws InstitutionCreationException, NotFoundException {
-        String institutionSlug = slugService.slugify(request.getName());
-        save(Institution.builder()
-                .name(request.getName())
-                .slug(institutionSlug)
-                .parent(null)
-                .address(request.getAddress())
-                .city(request.getCity())
-                .postalCode(request.getPostalCode())
-                .phone(request.getPhone())
-                .secondaryPhone(request.getSecondaryPhone())
-                .build());
-        uploadEmblemIfRequested(institutionSlug, request.getEmblem());
-
-        return institutionSlug;
+    public String create(CreateInstitutionRequest request) throws InstitutionCreationException {
+        Institution newInstitution = doSave(institutionMapper.requestToPreEntity(request));
+
+        uploadEmblemIfRequested(newInstitution.getSlug(), request.getEmblem());
+
+        return newInstitution.getSlug();
     }
 
     @Transactional(rollbackFor = InstitutionCreationException.class)
-    public Institution save(Institution institution) throws InstitutionCreationException {
-        try {
-            return institutionRepository.save(institution);
-        } catch (DataIntegrityViolationException e) {
-            log.info("Cannot create Institution with name '" + institution.getName() + "'", e);
-            throw new InstitutionCreationException("Cannot create Institution", e);
-        }
+    public String create(Institution institution) throws InstitutionCreationException {
```

Comment:
why this function have been renamed from `save` to `create`? Haven't used before `save` name of such methods?

Reply:
There are some methods with save, although most of the creational methods (among services) stem from create. Also, the child-creating method in InstService already had the create in name, so I homogenized the naming.